### PR TITLE
ref(sort): Hard code variant values

### LINKED ISSUE #48490: Hard code 3 combinations for EA
Hard code the 3 winning combinations from the internal testing for an [A/B/C test](https://develop.sentry.dev/ab-testing/) in EA.

### PR DESCRIPTION
**Pull request by ceorourke**
Set up the backend to read experiment values and change the sort variants. Note that these values are currently being tested and will change, I just wanted to get this ready.

Closes #48490

### CODE REVIEW

File: src/sentry/api/endpoints/organization_group_index.py [hunk 1]
```diff
@@ -226,7 +238,20 @@ def _search(
                 query_kwargs.update(extra_query_kwargs)
 
             if query_kwargs["sort_by"] == "betterPriority":
-                query_kwargs["aggregate_kwargs"] = self.build_better_priority_sort_kwargs(request)
+                if features.has(
+                    "organizations:better-priority-sort-experiment",
```
Comment (scefali) [inline src/sentry/api/endpoints/organization_group_index.py:242]: @ceorourke do we need to check `issue-list-better-priority-sort` to see if we should force to variant 1?

File: src/sentry/api/endpoints/organization_group_index.py [hunk 2]
```diff
@@ -226,7 +238,28 @@ def _search(
...
+                    organization,
+                    actor=request.user,
+                ):
+                    choice = expt_manager.get(
+                        "PrioritySortExperiment", org=organization, actor=request.user
+                    )
+                    # force into variant1 for internal testing
+                    if features.has(
```
Comment (scefali) [inline src/sentry/api/endpoints/organization_group_index.py:250]: @ceorourke You shouldn't need the experiment AND the feature flag, we should be ORing the feature flags, right?

File: src/sentry/api/endpoints/organization_group_index.py [hunk 3]
```diff
@@ -226,7 +238,29 @@ def _search(
...
+                    )
+                # force into variant1 for internal testing
+                if features.has(
+                    "organizations:issue-list-better-priority-sort",
+                    organization,
+                    actor=request.user,
+                ):
+                    choice = "variant1"
+
+                if choice == "baseline":
```
Comment (scefali) [inline src/sentry/api/endpoints/organization_group_index.py:258]: @ceorourke I think this needs to be an elif so we don't wipe out the variant1 override

File: src/sentry/api/endpoints/organization_group_index.py [hunk 4]
```diff
@@ -226,7 +238,29 @@ def _search(
...
+                    query_kwargs["sort_by"] = "date"
+                else:
+                    query_kwargs["aggregate_kwargs"] = self.build_better_priority_sort_kwargs(
```
Comment (scefali) [inline src/sentry/api/endpoints/organization_group_index.py:261]: @ceorourke do we need this if `organizations:issue-list-better-priority-sort` is true as well?

File: src/sentry/api/endpoints/organization_group_index.py (hunk 3 above)
Comment (ceorourke) [reply src/sentry/api/endpoints/organization_group_index.py:258]: What do you mean? It's just checking if the choice is baseline and setting the sort to last seen here. It's not overwriting the choice value.

Comment (scefali) [reply src/sentry/api/endpoints/organization_group_index.py:258]: @ceorourke but its going to make `query_kwargs["sort_by"] = "date"` which isn't what we want if the feature flag is enabled, right?

Comment (ceorourke) [reply src/sentry/api/endpoints/organization_group_index.py:258]: If they're in the experiment they could fall into baseline, variant1, or variant2. All I'm doing here is setting baseline to last seen.

Comment (scefali) [reply src/sentry/api/endpoints/organization_group_index.py:258]: @ceorourke right...but if they have the `organizations:issue-list-better-priority-sort` set to true then we want to force to variant1 and ignore the experiment value.

File: src/sentry/api/endpoints/organization_group_index.py (hunk 4 above)
Comment (ceorourke) [reply src/sentry/api/endpoints/organization_group_index.py:261]: Yeah? What do you mean? In the function that builds the kwargs it set the appropriate kwargs for variant1

File: src/sentry/api/endpoints/organization_group_index.py (hunk 3 above)
Comment (ceorourke) [reply src/sentry/api/endpoints/organization_group_index.py:258]: How does it not do that?

Comment (scefali) [reply src/sentry/api/endpoints/organization_group_index.py:258]: @ceorourke you're right and I'm bad at reading code

**Review by scefali — APPROVED** [review]

**Pull request merged**